src/js/base/runtime.js

```diff
@@ -5032,17 +5065,25 @@ function (Namespace, jsnums, codePoint, seedrandom, util) {
 
         return thisRuntime.safeCall(function() {
           if (mod.nativeRequires.length === 0) {
-            // CONSOLE.log("Nothing to load, skipping stack-pause");
             return mod.nativeRequires;
           } else {
-            return thisRuntime.pauseStack(function(restarter) {
-              // CONSOLE.log("About to load: ", mod.nativeRequires);
-              require(mod.nativeRequires, function(/* varargs */) {
-                var nativeInstantiated = Array.prototype.slice.call(arguments);
-                //CONSOLE.log("Loaded: ", nativeInstantiated);
-                restarter.resume(nativeInstantiated);
+            if (thisRuntime.bounceAllowed) {
+              /* Use async version of require() */
+              return thisRuntime.pauseStack(function(restarter) {
+                require(mod.nativeRequires, function(/* varargs */) {
+                  var nativeInstantiated = Array.prototype.slice.call(arguments);
+                  restarter.resume(nativeInstantiated);
+                });
               });
-            });
+            } else {
+              /* require() should be synchronous */
+              var arr = [];
+              for(var i = 0; i < mod.nativeRequires.length; i++) {
+                var val = require(mod.nativeRequires[i]);
```

Comment:
Making a note here that this requires that all native modules are not actually loaded asynchronously.  This rules out some uses cases (which we don't rely on at all, but theoretically could use with bounces allowed), where requirejs will fetch from a URL and asynchronously load.  There were past versions of the plot library with d3 that relied on this, for example.